Explorer/Assets/Scripts/Utility/Storage/PersistentSetting.cs

```diff
@@ -125,20 +133,58 @@ public T Value
         {
             get
             {
-                // It's for runtime only, in case it is used from the Editor it should not be saved anywhere
-                if (Application.isPlaying)
-                    return getValue!(key, defaultValue);
-
-                return defaultValue;
+                EnsureMainThread();
+                return getValue!(key, defaultValue)!;
             }
 
             set
             {
+                EnsureMainThread();
+
+                // It's for runtime only, in case it is used from the Editor it should not be saved anywhere
                 if (!Application.isPlaying)
                     return;
 
                 setValue!(key, value);
             }
         }
+
+        public CachedSetting<T, PersistentSetting<T>> WithCached() =>
+            new (this);
+
+        [Conditional("DEBUG")]
+        private void EnsureMainThread()
+        {
+            if (PlayerLoopHelper.IsMainThread == false)
+                throw new InvalidOperationException($"Cannot access PersistentSetting outside of the main thread, key: {key}, current thread: {Thread.CurrentThread.ManagedThreadId}, main thread: {PlayerLoopHelper.MainThreadId}");
+        }
+    }
+
+    public struct CachedSetting<T, TK> : ISetting<T> where TK: ISetting<T> where T: IEquatable<T>
```

Comment:
Without caching it would access PlayerPrefs on each request and it's bad for the performance. In this particular case it would access the hard drive on each WebRequest and we call WebRequests really often

Reply:
It's only true in the Editor.
For build it's not:

```
When you call PlayerPrefs.GetString("key") (or similar methods for other types), Unity retrieves the data from a cached version in memory if it exists.
If the data isn't already cached, Unity will read it from the storage where PlayerPrefs data is saved (e.g., the Windows Registry on Windows, a plist file on iOS, or a hidden file on Android).
```

Reply:
Thus, I think, you can remove `CachedSetting`